src/main/kotlin/traffic_simulation/Vehicle.kt

```diff
@@ -1,31 +1,10 @@
 package traffic_simulation
 
-class Vehicle(val id : Int, val wannaDrive : Boolean, var delayed : Boolean = false, var hasDriven : Boolean = false){
-    val allVehicles : MutableList<Vehicle> = mutableListOf()
+class Vehicle(val id : Int, val wannaDrive : Boolean){
+    var delayed : Boolean = false //no vehicle is delayed by default
 
-    fun isNewVehicle (vehicles : MutableList<Vehicle>) : Boolean {
-        var isNew : Boolean = true
-        for (vehicle in vehicles){
-            if (this.id == vehicle.id){
-                isNew = false
-            }
-        }
-        return isNew
-    }
-
-    fun newVehicle (vehicles : MutableList<Vehicle>): Vehicle{
-        if (isNewVehicle(vehicles)) {
-            vehicles.add(this)
-        }
-        return Vehicle(id, wannaDrive)
-    }
-
-    fun wannaDrive (): Boolean {
-        return wannaDrive
-    }
-
-    fun delayed (): Vehicle {
-        return Vehicle (id , wannaDrive , true)
+    fun gettingDelayed(){
+        delayed = true
```

Comment:
Wie oben beschrieben ist `delayed` ein Attribut, d.h. man kann das hier mit `this` setzen.